### PR TITLE
Removed double loop, updated tests

### PR DESCRIPTION
**Pull request by naterush**
Had to change a bit with tests, but it still looks pretty. Note: addresses now have to be sorted in increasing order w/in the array of signatures.

### CODE REVIEW

File: contracts/SimpleMultiSig.sol [hunk 1]
```diff
@@ -1,39 +1,34 @@
 pragma solidity 0.4.11;
 contract SimpleMultiSig {
 
-  uint public nonce; // mutable state
-  uint public threshold; // immutable state
-  address[] public owners; // immutable state
+  uint public nonce;                // (only) mutable state
+  uint public threshold;            // immutable state
+  mapping (address => bool) owners; // immutable state
```
Comment (christianlundkvist) [inline contracts/SimpleMultiSig.sol]: Maybe call this one `isOwner` for clarity?

File: test/simplemultisig.js [hunk 2]
```diff
@@ -15,15 +15,22 @@ contract('SimpleMultiSig', function(accounts) {
...
   let hash = solsha3(input)
 
   let sigV = []
   let sigR = []
   let sigS = []
 
+  //Put addresses in increasing order
+  if (doSort) {
+    signers.sort(function(a, b) {
+      return parseInt(a) - parseInt(b);
```
Comment (christianlundkvist) [inline test/simplemultisig.js]: Here I would just do `signers.sort()` and it should sort the strings correctly.

Comment (christianlundkvist) [inline test/simplemultisig.js]: In fact you can do that before even passing it into the function, and you may not need the extra `doSort` parameter.

File: contracts/SimpleMultiSig.sol (hunk 1 above)
Comment (naterush) [inline contracts/SimpleMultiSig.sol]: Good call

File: test/simplemultisig.js (hunk 2 above)
Comment (naterush) [inline test/simplemultisig.js]: Will remove the doSort param.

**Comment by naterush** [comment]
@christianlundkvist updated with your comments :)

**Comment by christianlundkvist** [comment]
Thanks, merging!

**Pull request merged**